server/src/main/java/org/apache/cassandra/sidecar/acl/authentication/AuthenticationProviderFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.acl.authentication;
+
+import java.util.Map;
+
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.handler.impl.AuthenticationHandlerInternal;
+import org.apache.cassandra.sidecar.config.AccessControlConfiguration;
+import org.apache.cassandra.sidecar.exceptions.ConfigurationException;
+
+public interface AuthenticationProviderFactory
+{
+    default void validate(Map<String, String> parameters) throws ConfigurationException
```

Comment:
`validate` should be part of `initialize/create` method, instead of a standalone step. Since both methods take a map as input, there is no guarantee (from the perspective of the interface) that the map used for creating the handler, is the same as the map for validation. 
I would suggest removing the `validate` method. The implementations run validation on the parameters anyway.

Reply:
Makes sense, was thinking the same.